src/Backend/Modules/Extensions/Engine/Model.php

```diff
@@ -516,17 +507,43 @@ public static function getModulesThatRequireAkismet(): array
      */
     public static function getModulesThatRequireGoogleMaps(): array
     {
-        $modules = [];
-        $installedModules = BackendModel::getModules();
+        return self::getModulesThatRequireSetting('google_maps');
+    }
 
-        foreach ($installedModules as $module) {
-            $setting = BackendModel::get('fork.settings')->get($module, 'requires_google_maps', false);
-            if ($setting) {
-                $modules[] = $module;
-            }
+    /**
+     * Fetch the list of modules that require Google Recaptcha API key
+     *
+     * @return array
+     */
+    public static function getModulesThatRequireGoogleRecaptcha(): array
+    {
+        return self::getModulesThatRequireSetting('google_recaptcha');
+    }
+
+    /**
+     * Fetch the list of modules that require a certain setting. The setting is affixed by 'requires_'
+     *
+     * @param string $setting
+     *
+     * @return array
+     */
+    private static function getModulesThatRequireSetting(string $setting): array
+    {
+        if ($setting === '') {
+            return [];
         }
 
-        return $modules;
+        /** @var ModulesSettings $moduleSettings */
+        $moduleSettings = BackendModel::get('fork.settings');
+
+        return array_filter(
+            BackendModel::getModules(),
+            function (string $module) use ($moduleSettings, $setting): bool {
+                $requiresGoogleRecaptcha = $moduleSettings->get($module, 'requires_' . $setting, false);
```

Comment:
The name of $requiresGoogleRecaptcha is wrong, since this is a common method.
`$modulesWhichRequireTheSetting` seems better